src/python/pants/core/util_rules/search_paths.py

```diff
@@ -142,5 +146,35 @@ async def validate_search_paths(request: ValidateSearchPathsRequest) -> Validate
     return ValidatedSearchPaths(search_paths)
 
 
+class ExecutableSearchPathsOptionMixin:
+    executable_search_paths_help: str
+    _options_env: EnvironmentVars
+
+    executable_search_paths = StrListOption(
+        default=["<PATH>"],
```

Comment:
It's probably best to name this as `_executable_search_paths` to direct consumers to the post-processed `@property`, unless you feel there's a good reason to make the raw version more visible.